src/Accelerators/NNPA/Conversion/ONNXToZHigh/RewriteONNXForZHigh.cpp

```diff
@@ -388,6 +388,62 @@ class SplitLargeMatMulPattern : public OpRewritePattern<ONNXMatMulOp> {
   }
 };
 
+/// This pattern is to replace `C = add/sub(A, B)` by `A` when B is a zero
```

Comment:
actually, we could move Add/SubZerosOnRhs and the similar patterns (MulOnes, MatMulZeros, MatMulIntegerConstZero) to here, as they don't do conventional constant propagation and folding

in fact, they are getting in the way in the constant propagation pass as I'm restructuring constant propagation to support "lazy" delayed, symbolic constant folding

Reply:
I see. Only one input is a constant. It looks canonicalization can be another candidate. I put this pattern here because it requires DimAnalyis, so it's beneficial to NNPA only. We should find a new home for it, e.g. a new onnx-to-onnx transformation that utilizes DimAnalysis.